src/coprocessor/codec/mysql/mod.rs

```diff
@@ -64,17 +64,25 @@ mod tests {
     #[test]
     fn test_parse_frac() {
         let cases = vec![
+            ("", 0, 0),
+            ("", 10, 0),
             ("1234567", 0, 0),
             ("1234567", 1, 1),
             ("0000567", 5, 6),
             ("1234567", 5, 12346),
             ("1234567", 6, 123457),
             ("9999999", 6, 1000000),
+            ("9999999", 8, 99999990),
```

Comment:
I don't think we need to support anything larger than MAX_FSP. Otherwise we'll need to support a test cases like `("", 9999, 0)` which obviously panics.